rust/ballista/src/bin/executor.rs

```diff
@@ -77,9 +77,11 @@ async fn poll_loop(
                     info!("Current task finished");
                     *running_task_guard = None;
                     task_status.push(TaskStatus {
-                        job_id: task_id.job_id,
-                        stage_id: task_id.stage_id,
-                        partition_id: task_id.partition_id,
+                        partition_id: Some(PartitionId {
+                            job_id: task_id.job_id,
+                            stage_id: task_id.stage_id,
+                            partition_id: task_id.partition_id,
+                        }),
```

Comment:
can't you do this? (maybe you'll need to clone)

```suggestion
                        partition_id: Some(task_id),
```

Reply:
🤦 definitely